src/net/sourceforge/kolmafia/session/LocketManager.java

```diff
@@ -0,0 +1,148 @@
+    return knownMonsters.contains(monsterId);
+  }
+
+  public static boolean foughtMonster(int monsterId) {
+    return foughtMonsters.contains(monsterId);
+  }
+
+  public static void parseMonsters(final String text) {
+    // Visiting the reminisce page is a source of truth
+    knownMonsters.clear();
+
+    // Add all monsters from the reminisce page
+    var m = REMINISCABLE_MONSTER.matcher(text);
+    while (m.find()) {
+      knownMonsters.add(Integer.parseInt(m.group(1)));
+    }
+
+    // Add all the monsters you've fought today, which will not otherwise show on said page
+    parseFoughtMonsters();
+    knownMonsters.addAll(foughtMonsters);
+  }
+
+  public static void parseFight(final MonsterData monster, final String text) {
+    if (!text.contains("loverslocketframe.png")) {
+      return;
+    }
+
+    // This will not double an existing id so is safe to run at any round
+    addFoughtMonster(monster.getId());
+
+    Preferences.setString("locketPhylum", monster.getPhylum().toString());
+  }
+
+  /**
+   * Work out the current locket phylum from the locket's item description
+   *
+   * @param responseText HTML for the item description to parse
+   */
+  public static void parseLocket(final String responseText) {
+    var mods = new Modifiers.ModifierList();
+
+    // The plan here is to parse the locket description...
+    DebugDatabase.parseItemEnchantments(
+        responseText, mods, new ArrayList<>(), KoLConstants.EQUIP_ACCESSORY);
+
+    // ...find the first modifier that can indicate the phylum...
+    var indicativeMod =
+        mods.stream()
+            .map(Modifiers.Modifier::getName)
+            .filter(Predicate.not(CONSTANT_MODS::contains))
+            .findAny()
+            .orElse(null);
+
+    // ... grab the raw mod string for the locket from modifiers.txt...
+    var locketModString =
+        Modifiers.getItemModifiers(ItemPool.COMBAT_LOVERS_LOCKET).getString(Modifiers.MODIFIERS);
+
+    String phylum = "";
+
+    // ... and the locket has a phylum at all...
+    if (indicativeMod != null) {
+      // Weapon Damage: [pref(locketPhylum,weird)*25]
+      int start = locketModString.indexOf(indicativeMod) + indicativeMod.length() + 21;
+      // ... we find the indicative modifier in the raw mod string and see what phylum it is
+      // associated with.
+      phylum = locketModString.substring(start, locketModString.indexOf(")", start));
+    }
+
+    Preferences.setString("locketPhylum", phylum);
+  }
+
+  public static boolean own() {
+    return KoLConstants.inventory.contains(ItemPool.get(ItemPool.COMBAT_LOVERS_LOCKET, 1))
+        || KoLCharacter.hasEquipped(ItemPool.COMBAT_LOVERS_LOCKET);
+  }
+
+  public static void clear() {
+    knownMonsters.clear();
+    foughtMonsters.clear();
+  }
+
+  public static void reset() {
+    if (!own()) {
+      clear();
+      return;
+    }
+
+    RequestThread.postRequest(new GenericRequest("inventory.php?reminisce=1", false));
```

Comment:
Does anyone know how I stop this printing "Encounter: <encountername>" to the CLI and session log? I can see how I'd do it, but I'm highly sceptical that there's no other special casing for encounters. I'm sure this is not the first encounter we want to visit silently...

Reply:
@Veracity0 do you have any thoughts?

Reply:
That looks like it, thanks @midgleyc!